libraries/splunk_app.rb

```diff
@@ -126,13 +174,80 @@ def create_splunk_directory(path)
         dir.group(node['splunk']['group'])
         dir.mode('0755')
         dir.run_action(:create)
-        new_resource.updated_by_last_action(dir.updated_by_last_action?)
+        new_resource.updated_by_last_action(true) if dir.updated_by_last_action?
+      end
+
+      def download_and_install # rubocop:disable PerceivedComplexity, CyclomaticComplexity
+        return if new_resource.url.nil? || new_resource.url.empty?
+
+        expected_version = CernerSplunk::AppVersion.new new_resource.version
+        installed_version = CernerSplunk::AppVersion.new @current_resource.version
+
+        return if expected_version.type && expected_version == installed_version
+        Chef::Log.warn "Expected version not specified for #{new_resource.app}." unless expected_version.type
+        fail "Expecting to install prerelease on top of same released version for #{new_resource.app}" if expected_version.type == :prerelease && installed_version.type == :base && expected_version.base == installed_version.base
+
+        # In whyrun mode, the rest is not safe.
```

Comment:
That makes sense.  Do you mind adding a commit to that effect here?  Something like: 

```rb
# When in whyrun mode, stop so we do not modify the node by adding a tarball.
```